Sources/HttpPipeline/Middleware.swift

```diff
@@ -97,7 +98,7 @@ public func send(_ data: Data?) -> Middleware<BodyOpen, BodyOpen, Data?, Data?>
     }
 
     return .init(
-      data: conn.data,
+      data: concatenatedData,
```

Comment:
Failure to set this prevented `send` from working. Not sure if this is the right fix though? Feeling our data model may not be quite what we want it to be here.